packages/serve/src/lib/middleware/accessLogMiddleware.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  getLogger,
+  LoggerOptions,
+  VulcanInternalExtension,
+} from '@vulcan-sql/core';
+import * as bytes from 'bytes';
+import { BuiltInMiddleware, KoaContext, Next } from '@vulcan-sql/serve/models';
+
+@VulcanInternalExtension('access-log')
+export class AccessLogMiddleware extends BuiltInMiddleware<LoggerOptions> {
+  private logger = getLogger({
+    scopeName: 'ACCESS_LOG',
+    options: this.getOptions(),
+  });
+
+  public async handle(context: KoaContext, next: Next) {
+    if (!this.enabled) return next();
+
+    const { request: req, response: resp, params } = context;
+
+    const reqSize = req.length ? bytes(req.length).toLowerCase() : 'none';
+    const respSize = resp.length ? bytes(resp.length).toLowerCase() : 'none';
```

Comment:
Nice to print request/response size, but I can get both of them in lab env, how do we get them to work?

Should we calculate the response size after other middlewares (after next() function)? Response data might be set after them.

Reply:
Hi @oscar60310, thanks for reviewing and suggesting, I have checked the lab env both, and like you said the request/response size (Content-Length) shows an `undefined` value ( Also includes `integrating-testing` and my `app.spec` test cases in `serve` package.

So I searching for the reason, and I the request header only have `Content-Length` when HTTP method is `POST` or `PUT`, because `Content-Length` only calculate payload data size, please check the references below and my test by sample with `typescript-koa-starter`:
- https://www.ibm.com/docs/en/ibm-mq/9.0?topic=headers-content-length-http-entity-header
- https://stackoverflow.com/questions/8540931/what-is-the-correct-content-length-to-set-for-a-get-request
- https://stackoverflow.com/questions/3854842/content-length-header-with-head-requests

**POST request and return JSON format:**
<img width="844" alt="POST-JSON" src="https://user-images.githubusercontent.com/5389253/190344171-54d08439-5a1d-4139-ac6e-4b6d25f20f85.png">


**GET request and return JSON format:**
<img width="841" alt="GET-JSON" src="https://user-images.githubusercontent.com/5389253/190344106-931cef1a-3c0e-479b-8a05-97f2629720ed.png">

For the response header, as we know, because we use the stream and the header will shows `Transfer-Encoding: chunck`, so `Content-Length` won't work.

- https://stackoverflow.com/questions/2773396/whats-the-content-length-field-in-http-header

**GET request and return Stream format:**
<img width="844" alt="GET-STREAM" src="https://user-images.githubusercontent.com/5389253/190344724-19497ca6-99ef-4c91-aeaa-37a4a82f1a1e.png">

**POST request and return Stream format:** 
<img width="841" alt="POST-STREAM" src="https://user-images.githubusercontent.com/5389253/190345267-85ab2476-ffa5-48a9-81b1-cb6eb49edf3a.png">

Reply:
Currently I still keep the request and response size, because maybe we will have POST / PUT API in the future.

Reply:
Thanks for the survey!